Documentation/api-guidelines/System.Memory.md

```diff
@@ -0,0 +1,46 @@
+# System.Memory Design Guidelines
+
+`System.Memory` is a collection of types and features that make working with
+buffers and raw memory more efficient while remaining type safe. The feature
+specs can be found here:
+
+* [`Span<T>`](https://github.com/dotnet/corefxlab/blob/master/docs/specs/span.md)
+* [`Memory<T>`](https://github.com/dotnet/corefxlab/blob/master/docs/specs/memory.md)
+
+## Overview
+
+* `ReadOnlySpan<T>` is effectively the universal receiver, in that `T[]`, `T*`,
+  `Memory<T>`, `ReadOnlyMemory<T>`, `Span<T>`, `ArraySegment<T>` can all be
+  converted to it.  So if you can declare your API to accept a `ReadOnlySpan<T>`
+  and behave efficiently, that's best, as any of these inputs can be used with
+  your method.
+* Similarly for `Span<T>`, if you need write access in the implementation.
+* It allows building safe public APIs that can operate on unmanaged memory
+  without forcing all consumers to use pointers (and thus becoming unsafe) while
+  still getting equivalent performance.
```

Comment:
> while still getting equivalent performance

This isn't necessarily true. The safety/convenience comes with some cost. @jkotas

Reply:
Yeah, I think we can say that Span\<T\> is on par with T[], not with T*.

Reply:
I worded that badly. What I meant to say is: the method can use pointers in the implementation but accepts `Span<T>` as the argument. I'll reword.